ui/src/app/machines/components/HeaderStrip/ActionFormWrapper/ActionFormWrapper.js

```diff
@@ -0,0 +1,143 @@
+    case "lock":
+      return `${machineString} cannot be locked`;
+    case "override-failed-testing":
+      return `Cannot override failed tests on ${machineString}.`;
+    case "rescue-mode":
+      return `${machineString} cannot be put in rescue mode`;
+    case "set-pool":
+      return `Cannot set pool of ${machineString}`;
+    case "set-zone":
+      return `Cannot set zone of ${machineString}`;
+    case "unlock":
+      return `${machineString} cannot be unlocked`;
+    default:
+      return `${machineString} cannot be ${action.sentence}`;
+  }
+};
+
+export const ActionFormWrapper = ({
+  selectedAction,
+  selectedMachines,
+  setSelectedAction,
+  setSelectedMachines
+}) => {
+  const [actionableMachines, setActionableMachines] = useState([]);
+  const actionDisabled = actionableMachines.length !== selectedMachines.length;
+
+  useLayoutEffect(() => {
+    if (selectedAction) {
+      const actionable = selectedMachines.filter(machine =>
+        machine.actions.includes(selectedAction.name)
+      );
+      setActionableMachines(actionable);
+    }
+  }, [selectedAction, selectedMachines]);
+
+  useEffect(() => {
+    if (selectedMachines.length === 0) {
+      setSelectedAction(null);
+    }
+  }, [selectedMachines, setSelectedAction]);
+
+  return (
+    <Row>
+      <hr />
+      <Col size="12">
+        {actionDisabled && (
+          <p data-test="machine-action-warning">
+            <i className="p-icon--warning" style={{ marginRight: ".5rem" }} />
+            <span>
+              {getErrorSentence(
+                selectedAction,
+                selectedMachines.length - actionableMachines.length
+              )}
+              . To proceed,{" "}
+              <Button
+                appearance="link"
+                data-test="select-actionable-machines"
+                inline
+                onClick={() => setSelectedMachines(actionableMachines)}
+              >
+                update your selection
+              </Button>
+              .
+            </span>
+          </p>
+        )}
+        {!actionDisabled && (
+          // TODO: Replace with form components for each action.
+          <div className="u-align--right">
+            <Button
+              appearance="neutral"
+              data-test="cancel-action"
+              onClick={() => setSelectedAction(null)}
+              style={{ marginBottom: ".4rem" }}
+            >
+              Cancel
+            </Button>
+            <Button
+              appearance={
+                selectedAction.name === "delete" ? "negative" : "positive"
+              }
+              style={{ marginBottom: ".4rem" }}
+              type="submit"
+            >
+              {getSubmitText(selectedAction, selectedMachines.length)}
+            </Button>
+          </div>
+        )}
+      </Col>
+    </Row>
+  );
+};
+
+ActionFormWrapper.propTypes = {
+  selectedAction: PropTypes.object, // TODO: PropType shapes for domain entities
```

Comment:
Are these todos captured somewhere? If not could be good to add issues.

Reply:
Good call. Added here: https://github.com/canonical-web-and-design/maas-ui/issues/826